### PR TITLE
feat: banners can now be uploaded to bots

### PR DESCRIPTION
**Pull request by Jaskowicz1**
This PR adds the ability to add banners to bots. I have also removed the limit imposed on avatars and banners as it seems the 256kb limit isn't true.

## Code change checklist

- [x] I have ensured that all methods and functions are **fully documented** using doxygen style comments.
- [x] My code follows the [coding style guide](https://dpp.dev/coding-standards.html).
- [x] I tested that my change works before raising the PR.
- [x] I have ensured that I did not break any existing API calls.
- [x] I have not built my pull request using AI, a static analysis tool or similar without any human oversight.

### CODE REVIEW

**Comment by Jaskowicz1** [comment]
forgor to `cmake -B build` lol

**Comment by Jaskowicz1** [comment]
Converted to draft, need to experiment with file limits as it seems avatars and banners do not have a 256kb constraint.

**Comment by Jaskowicz1** [comment]
No longer draft!

**Comment by Jaskowicz1** [comment]
I found that banners and avatars have no limit for bots? It could possibly be 25MB but Discord doesn't document this and it seems the request times out at anything over 9mb (at least it did for me). I assume it's a combination of can you upload in time and can Discord process it in time to respond, otherwise it throws a malformed HTTP request.

So, this PR now removes the limit for avatars and banners.

File: include/dpp/cluster.h [hunk 1]
```diff
@@ -3253,19 +3253,24 @@ class DPP_EXPORT cluster {
...
+	 *
+	 * @note There appears to be no limit to the image size, however, if your image cannot be processed/uploaded in time, you will receive a malformed http request.
 	 *
-	 * Modifies the current member in a guild. Returns the updated guild_member object on success.
-	 * Fires a `Guild Member Update` Gateway event.
 	 * @see https://discord.com/developers/docs/resources/user#modify-current-user
 	 * @param nickname Nickname to set
-	 * @param image_blob Avatar data to upload (NOTE: Very heavily rate limited!)
-	 * @param type Type of image for avatar. It can be one of `i_gif`, `i_jpg` or `i_png`.
+	 * @param avatar_blob Avatar data to upload
```
Comment (braindigitalis) [inline include/dpp/cluster.h:3265]: documentation seems to be duplicated?

File: include/dpp/cluster_sync_calls.h [hunk 2]
```diff
@@ -2765,23 +2765,28 @@ confirmation thread_member_remove_sync(snowflake thread_id, snowflake user_id);
...
+ * @note There appears to be no limit to the image size, however, if your image cannot be processed/uploaded in time, you will receive a malformed http request.
  *
- * Modifies the current member in a guild. Returns the updated guild_member object on success.
- * Fires a `Guild Member Update` Gateway event.
  * @see dpp::cluster::current_user_edit
  * @see https://discord.com/developers/docs/resources/user#modify-current-user
  * @param nickname Nickname to set
- * @param image_blob Avatar data to upload (NOTE: Very heavily rate limited!)
- * @param type Type of image for avatar. It can be one of `i_gif`, `i_jpg` or `i_png`.
+ * @param avatar_blob Avatar data to upload
```
Comment (braindigitalis) [inline include/dpp/cluster_sync_calls.h:2778]: docs duplicated

**Review by braindigitalis — APPROVED** [review]

**Pull request merged**